wordpress-mobile/WordPress-iOS
# Save for Later: update reader menu view model

Implements #9229 

This PR adds a section to the Reader headers to present the Saved for Later / Bookmarked items.

![bookmarks](https://user-images.githubusercontent.com/2722505/39510681-1cab06e6-4e1d-11e8-855b-96640d1cf485.jpg)

In the spirit of increasing testability and modularity, I have taken the opportunity to extract some logic out of `ReaderMenuViewController`, also with the intention of making the current implementation a bit more open for modification.

Caveats:
- In order to present the menu item, I have added a mock subclass of `ReaderAbstractTopic` called `ReaderSaveForLaterTopic`. This class should be a proper core data entity. and will have to be updated when the core data model is.
- Also, there is some hardcoded logic in `ReaderMenuViewModel` hidden behind a feature flag, that is marked as "to be removed" after the new topic is provided by the core data model.
- I am terrible at naming stuff. Also, @malinajirka has suggested that we should probably use "Bookmark/ed" instead of "Saved for Later", which I think makes sense. But in the meantime, here we are! 😛 

On the bright side... there are some unit tests!

To test:
1. Checkout the branch
2. Run it as it is, the extra menu item should be visible in `localDeveloper` mode.
